src/replaceAll.js

```diff
@@ -1,34 +1,33 @@
-import { curryN, split, join, pipe } from 'ramda';
+import { curryN } from 'ramda';
 
-import isRegExp from './isRegExp';
+import polyfill from './internal/polyfills/String.replaceAll';
+import isFunction from './isFunction';
 
 /**
  * Replaces all substring matches in a string with a replacement.
  *
  * @func replaceAll
  * @memberOf RA
- * @since {@link https://char0n.github.io/ramda-adjunct/2.17.0|v2.17.0}
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}
  * @category String
  * @sig String -> String -> String -> String
- * @param {string} searchValue The substring to match
+ * @param {string} searchValue The substring or a global RegExp to match
  * @param {string} replaceValue The string to replace the matches with
  * @param {string} str The String to do the search and replacement in
  * @return {string} A new string containing all the `searchValue` replaced with the `replaceValue`
- * @throws {Error} When `searchValue` is RegExp
  * @see {@link http://ramdajs.com/docs/#replace|R.replace}, {@link https://github.com/tc39/proposal-string-replaceall|TC39 proposal}
  * @example
  *
  * RA.replaceAll('ac', 'ef', 'ac ab ac ab'); //=> 'ef ab ef ab'
+ * RA.replaceAll('', '_', 'xxx'); //=> '_x_x_x_'
+ * RA.replaceAll(/x/g, 'v', 'xxx'); //=> 'vvv'
+ * RA.replaceAll(/x/, 'v', 'xxx'); //=> TypeError
  */
 const replaceAll = curryN(3, (searchValue, replaceValue, str) => {
```

Comment:
We have a convention that standardizes the way we use polyfills. Take a look at [isInteger](https://github.com/char0n/ramda-adjunct/blob/6e9d1a91a26e45ac923ea3a90cab8274e9d6222d/src/isInteger.js#L27) and how to create [tests](https://github.com/char0n/ramda-adjunct/blob/6e9d1a91a26e45ac923ea3a90cab8274e9d6222d/test/isInteger.js#L8-L7) for it.

Reply:
@char0n, I hopeI get it right (I made a wrapper for standard .replaceAll to save RA.replaceAll API and curried it). Also I changed the polyfill API to match existing RA.ReplaceAll.